broker/src/main/java/io/camunda/zeebe/broker/system/management/BrokerAdminServiceImpl.java

```diff
@@ -134,39 +134,24 @@ private CompletableFuture<PartitionStatus> getPartitionStatus(final ZeebePartiti
             partitionStatus.completeExceptionally(error);
             return;
           }
-          if (currentRoleFuture.join() == Role.LEADER) {
-            final var streamProcessor = streamProcessorFuture.join();
-            final var exporterDirector = exporterDirectorFuture.join();
-            if (streamProcessor.isEmpty() || exporterDirector.isEmpty()) {
-              partitionStatus.completeExceptionally(
-                  new IllegalStateException(
-                      "No streamprocessor or exporter found for leader partition."));
-            } else {
-              getLeaderPartitionStatus(
-                  partition, streamProcessor.get(), exporterDirector.get(), partitionStatus);
-            }
-
+          final var role = currentRoleFuture.join();
+          final var streamProcessor = streamProcessorFuture.join();
+          final var exporterDirector = exporterDirectorFuture.join();
+
+          if (streamProcessor.isEmpty() || exporterDirector.isEmpty()) {
```

Comment:
:upside_down_face: I know it was this way before but I think it'd be nice to have separate error messages here. If I'd read "No streamprocessor or exporter found for leader partition" I'd also want to know which one didn't exist.

Reply:
Similarly, the message is no longer specific to leaders so it should read something like "No StreamProcessor found for partition %s"